packages/bitcore-node/src/services/p2p/index.ts

```diff
@@ -207,101 +226,89 @@ export class P2pRunner {
         });
         counter += 1;
         if (Date.now() - lastLog > 100) {
-          logger.info(`Sync progress ${(counter * 100 / goalHeight).toFixed(3)}%`, {
-            chain: this.chain,
-            network: this.network,
-            height: counter
-          });
+          logger.info(
+            `Sync progress ${(counter * 100 / goalHeight).toFixed(3)}%`,
+            {
+              chain: this.chain,
+              network: this.network,
+              height: counter
+            }
+          );
           lastLog = Date.now();
         }
         if (hash === finalHash) {
           start();
-        }
-        else {
+        } else {
           recentHash = hash;
         }
       },
-      start,
+      start
     };
   }
 }
 
-
-export class P2pProxy implements CSP.Provider<P2pRunner> {
-  private services: { [key: string]: P2pRunner };
-
-  constructor() {
-    this.services = {};
-  }
-
-  get({ chain }: Chain): P2pRunner {
-    if (this.services[chain]) {
-      return this.services[chain];
+export class P2pProxy implements CSP.Provider<Class<StandardP2p>> {
+  get({ chain }: Chain): Class<StandardP2p> {
+    if (P2PClasses[chain]) {
+      return P2PClasses[chain];
     }
     throw new Error(`Chain ${chain} doesn't have a P2P Worker registered`);
   }
 
-  register(chain: string, service: P2pRunner) {
-    this.services[chain] = service;
+  register(chain: string, service: Class<StandardP2p>) {
```

Comment:
Even if this compiles its semantics aren't quite right, `P2PClasses` has a type of:
```typescript
{ [key in keyof typeof SupportedChainSet]: Class<StandardP2p> }
```
So you cannot add `chain: string` to it.